src/main/java/org/broadinstitute/hellbender/tools/exome/segmentation/ScalarHMMSegmenter.java

```diff
@@ -0,0 +1,98 @@
+    }
+
+    // filter out components that have low weight and are too close to another component -- these will
+    // die out eventually in EM, but very slowly, so we hasten their demise for quicker convergence
+    @Override
+    protected void pruneUnusedComponents() {
+        final int K = weights.length;
+
+        final Set<Integer> componentsToPrune = new TreeSet<>();
+        for (final int state : IntStream.range(0, K).filter(n -> n != NEUTRAL_VALUE_INDEX).toArray()) {
+            final int closestOtherState = MathUtils.maxElementIndex(IntStream.range(0, K)
+                    .mapToDouble(n -> n == state ? Double.NEGATIVE_INFINITY : -Math.abs(hiddenStateValues.get(n) - hiddenStateValues.get(state)))
+                    .toArray());
+            final boolean hasLowWeight = weights[state] < MAX_WEIGHT_CONSIDERED_FOR_PRUNING;
+            final boolean isCloseToNeighbor = Math.abs(hiddenStateValues.get(state) - hiddenStateValues.get(closestOtherState)) < DISTANCE_TO_NEIGHBOR_TO_BE_CONSIDERED_SPURIOUS;
+            final boolean isVeryCloseToNeighbor = Math.abs(hiddenStateValues.get(state) - hiddenStateValues.get(closestOtherState)) < DISTANCE_TO_NEIGHBOR_TO_BE_CONSIDERED_DEFINITELY_SPURIOUS;
+
+            final boolean hasLessWeightThanNeighbor = weights[state] < weights[closestOtherState];
+
+            final boolean tinyWeight = weights[state] < AUTOMATICALLY_PRUNED_WEIGHT;
+            final boolean parasite = hasLowWeight && isCloseToNeighbor && hasLessWeightThanNeighbor;
+            final boolean isClone = isVeryCloseToNeighbor && hasLessWeightThanNeighbor;
+            if ( tinyWeight || parasite || isClone) {
+                componentsToPrune.add(state);
+            }
+        }
+
+        weights = IntStream.range(0, K)
+                .filter(n -> !componentsToPrune.contains(n)).mapToDouble(n -> weights[n]).toArray();
+        hiddenStateValues = IntStream.range(0, K)
+                .filter(n -> !componentsToPrune.contains(n)).mapToObj(hiddenStateValues::get).collect(Collectors.toList());
+
+        final StringBuilder message = new StringBuilder("After pruning (state, weight) pairs are: ");
+        for (int n = 0; n < weights.length; n++) {
+            final double weight = weights[n];
+            message.append(String.format("(%f, %f)", hiddenStateValues.get(n), weights[n]) + ((n < weights.length - 1) ? "; " : "."));
+        }
+        logger.info(message);
+    }
+
+    private static double[] uniformWeights(final int numStates) {
+        return Collections.nCopies(numStates, 1.0/numStates).stream().mapToDouble(x->x).toArray();
+    }
+
+    @Override
+    protected void relearnHiddenStateValues(final ExpectationStep eStep) {
+        final ClusteringGenomicHMM<DATA, Double> model = makeModel();
+        // by convention, state = 0 represents the neutral value (minor allele fraction = 1/2 or copy ratio = 1)
```

Comment:
Not sure if I see these changes.  To clarify, the constructor of `ScalarHMMSegmenter` could take a parameter `List<Integer> indicesOfPermanentHiddenStates` that indicates the indices of the states in `List<Double> hiddenStates` to ignore during pruning/relearning.  (Or `List<Integer> indicesOfConstantHiddenStates, or perhaps you can think of a better name!)  This would obviate the need for a static CONSTANT_VALUE_INDEX and the awkward conventions placed on `List<Double> hiddenStates`.